srcopsmetrics/entities/readme.py

```diff
@@ -32,31 +34,40 @@
 class ReadMe(Entity):
     """GitHub ReadMe entity."""
 
-    entity_schema = Schema({"name": str, "path": str, "content": str, "type": str, "license": str, "size": int,})
+    entity_schema = Schema({"name": str, "path": str, "content": str, "type": str, "size": int})
 
     def analyse(self) -> List[GithubContentFile]:
         """Override :func:`~Entity.analyse`."""
         # TODO: recursive Readme analysis - is that a good idea?
+        readme = self.get_raw_github_data()
 
         if self.previous_knowledge is None or len(self.previous_knowledge) == 0:
-            return [self.get_raw_github_data()]
+            return [readme]
 
-        if self.previous_knowledge["readme"]["size"] == self.repository.get_readme().size:
+        if not readme or (
+            "README" in self.previous_knowledge
+            and readme.decoded_content.decode("utf-8") == self.previous_knowledge["README"]["content"]
+            and readme.path == self.previous_knowledge["README"]["path"]
```

Comment:
Do wou want to add a warning here whe this happen?

Reply:
No, the Logger side is handled in iterator, where you are informed that no new knowledge was found to update